### PR TITLE
[KRC-51] input field

### PR DESCRIPTION
**Pull request by lucianojss**
[#KRC-51](https://digitalrig.atlassian.net/browse/KRC-51)

# Description 
Input text field with label and error.

Screenshot 2019-04-05 at 14 46 39

Screenshot 2019-04-05 at 14 46 32

### CODE REVIEW

File: src/components/Input/__tests__/Input.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { create } from 'react-test-renderer';
+
+import Input from '../Input.js';
+
+describe('COMPONENT - Input', () => {
+  describe('Input without error', () => {
+    let component;
+
+    beforeEach(() => {
```
Comment (blurbyte) [inline src/components/Input/__tests__/Input.test.js]: I would skip this **beforeEach** and **afterEach** declarations and just for clarity declare props in it block:
```
it('renders correctly', () => {
  const props = {
    ...
  };
  
  expect...
});
```

**Review by blurbyte — CHANGES_REQUESTED** [review]
Well done, looks good. The only thing I would change is actually split this component into 2. I feel it's trying to do too much.
First would be basic **Input** component which could be used to create Search, etc. in future.
And second one **FormInput** with error displayed.

File: src/components/Input/__tests__/Input.test.js (hunk 1 above)
Comment (lucianojss) [reply src/components/Input/__tests__/Input.test.js]: Done

**Comment by lucianojss** [comment]
> Well done, looks good. The only thing I would change is actually split this component into 2. I feel it's trying to do too much.
> First would be basic **Input** component which could be used to create Search, etc. in future.
> And second one **FormInput** with error displayed.

@blurbyte I followed your suggestion. It's ready again for review 👍

File: src/components/FormInput/FormInput.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const ErrorLabel = styled.div`
+  font-style: italic;
+  color: ${colorRed};
+`;
+
+const InputForm = ({ className, label, error, ...props }) => {
+  return (
+    <>
+      {label ? (
```
Comment (blurbyte) [inline src/components/FormInput/FormInput.js]: Due to accessibility requirements form label should not be optional.

File: src/components/FormInput/FormInput.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      {error && <ErrorLabel>{error}</ErrorLabel>}
+    </>
+  );
+};
+
+InputForm.propTypes = {
+  label: PropTypes.string,
+  error: PropTypes.string,
+  type: PropTypes.string.isRequired,
+  className: PropTypes.any
```
Comment (blurbyte) [inline src/components/FormInput/FormInput.js]: It's just string.

**Review by blurbyte — CHANGES_REQUESTED** [review]
Please make sure codecov.yaml is in sync with master.

File: src/components/FormInput/FormInput.js (hunk 3 above)
Comment (lucianojss) [reply src/components/FormInput/FormInput.js]: good catch, completely forgot to fix this

File: src/components/FormInput/FormInput.js (hunk 2 above)
Comment (lucianojss) [reply src/components/FormInput/FormInput.js]: Ok, I will make it required and remove the ternary validation.

Comment (blurbyte) [reply src/components/FormInput/FormInput.js]: 👍

**Review by blurbyte — APPROVED** [review]
Well done, let's merge it. In future please use regular functions at top level (not arrow) as in Input/Input.js

**Pull request merged**